### PR TITLE
[Defensive] FindCommand

### LINKED ISSUE #163: [Defensive] Find Command

### PR DESCRIPTION
**Pull request by underthehai**
Include check to ensure wishlist can only find activity object
Assert valid inputs
close #163

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
     public static final String MESSAGE_ACTIVITIES_LISTED_OVERVIEW = "%1$d activities listed!";
     public static final String MESSAGE_ACCOMMODATIONS_LISTED_OVERVIEW = "%1$d accommodations listed!";
     public static final String MESSAGE_FRIENDS_LISTED_OVERVIEW = "%1$d friends listed!";
-    public static final String MESSAGE_TRAVELPLANS_LISTED_OVERVIEW = "%1$d travel plans listed!";
+    public static final String WRONG_DIRECTORY = "You are currently in the wishlist which can only find activity.\n"
+            + "Please goto to a travelplan to edit friend, activity or accommodation";
```
Comment (jeannetoh99) [inline src/main/java/seedu/address/commons/core/Messages.java:24]: Is this a clearer phrasing?
"Please go to your desired travelplan to edit friend, activity or accommodation, using the goto command.";

File: src/main/java/seedu/address/logic/command/FindCommand.java [hunk 2]
```diff
@@ -22,52 +26,62 @@
...
+    public static final String MISSING_KEYWORDS = "Please provide at least one keyword to search for";
 
     private final NameContainsKeywordsPredicate predicate;
     private final int travelPlanObjectType;
 
     /**
      * Constructor for FindCommand
      *
-     * @param predicate contains a list of strings to filter travel plan object list
+     * @param predicate            contains a list of strings to filter travel plan object list
```
Comment (jeannetoh99) [inline src/main/java/seedu/address/logic/command/FindCommand.java:43]: I think you accidentally added a big space here haha

File: src/main/java/seedu/address/logic/command/FindCommand.java [hunk 3]
```diff
@@ -22,52 +26,62 @@
...
 
-        } else if (this.travelPlanObjectType == 2) { //friend
+        } else if (this.travelPlanObjectType == ParserUtil.FRIEND_INDEX) {
             model.updateFilteredFriendList(predicate);
             message = MESSAGE_FRIENDS_LISTED_OVERVIEW;
             size = model.getFilteredFriendList().size();
 
+        } else {
+            assert false;
         }
```
Comment (jeannetoh99) [inline src/main/java/seedu/address/logic/command/FindCommand.java:85]: Can use switch case but not a big problem.

**Review by jeannetoh99 — APPROVED** [review]
LGTM, just some comments!

File: src/main/java/seedu/address/logic/command/FindCommand.java (hunk 2 above)
Comment (lyeyixian) [inline src/main/java/seedu/address/logic/command/FindCommand.java:43]: Why is there a huge whitespace here tho?

**Review by jiaweiteo — APPROVED** [review]
LGTM

**Pull request merged**